lib/tessel/deploy.js

```diff
@@ -393,10 +403,144 @@ actions.glob = {
   }
 };
 
+actions.resolveBinaryModules = function(opts) {
+  var cwd = process.cwd();
+  var target = opts.target || cwd;
+  var relative = path.relative(cwd, target);
+  var globRoot = relative || target;
+
+  binaryModulesUsed.clear();
+
+  return new Promise((resolve, reject) => {
+    // Find all modules that include a compiled binary
+    var pattern = 'node_modules/**/*.node';
```

Comment:
> An easy way to detect them (without looking in the package.json like I am) is looking for a binding.gyp file. I'll move that code into a module eventually, but it's not a lot of code.

That's fine for detecting modules that will create a *.node binary, but if that were the metric here, we'd still have to resolve the path to the *.node (for replacement) later. This approach means that the resolution is done once, up front.